src/arith/const_int_bound.cc

```diff
@@ -371,7 +371,7 @@ class ConstIntBoundAnalyzer::Impl
     } else if (op->op.same_as(tir::builtin::bitwise_and())) {
       return VisitBitwiseAnd(op);
     } else if (op->op.same_as(tir::builtin::vscale()) && TargetHasSVE()) {
-      return MakeBound(1, 16);
+      return MakeBound(1, kAArch64VScaleValues.size());
```

Comment:
This is a lot of great work which will also serve as a nice foundation and example for the rest of the SME schedules. Thanks, @lhutton1! 

I just have a quick comment about the bounds of the "vscale" builtin. I understand that here we are making the assumption that the length of the vector containing all possible vscale values will be equal to the vector's maximum value (i.e. the vector will include all elements 0 < i <= max_vscale). Could an issue arise in the future from someone modifying the list of values without being aware of this assumption?

For example, I see other sources that mention vscale taking only powers of 2 as possible values:
- [LLVM](https://llvm.org/docs/LangRef.html#llvm-vscale-intrinsic:~:text=This%20function%20attribute%20indicates%20vscale%20is%20a%20power%2Dof%2Dtwo%20within%20a%20specified%20range): 

> This function attribute indicates vscale is a power-of-two within a specified range. 
- [Reference manual](https://developer.arm.com/documentation/ddi0487/latest/):

> For each IMPLEMENTATION DEFINED vector length, all the following apply:
> • The smallest architecturally defined maximum length is 128 bits.
> • The largest architecturally defined maximum length is 2048 bits.
> • The length is a power of two.

This might mean checking just the 1, 2, 4, 8, 16 values could be enough (slightly speeding up the analyzer), in which case the length of the `kAArch64VScaleValues` vector would no longer match its largest element. However, I'm happy to discuss and maybe address this in a separate PR later since it doesn't have anything to do with the rest of the schedule :)

Reply:
Thanks for taking a look @Anndrey24. I chose to use `.size()` here over something like `std::max_element()` due to speed (although admittedly it probably doesn't have much of an impact) in the analyzer, but I agree with your comments that it won't be the most maintainable going forwards. 

Regarding the range of values, I think you're right. I'm trying to recall the reason for going with 1, .., 16 but that escapes me for now. I think it would be best to discuss this further in another PR.